core/src/main/scala/akka/persistence/r2dbc/state/scaladsl/DurableStateDao.scala

```diff
@@ -458,6 +464,30 @@ private[r2dbc] class DurableStateDao(settings: R2dbcSettings, connectionFactory:
     Source.futureSource(result.map(Source(_))).mapMaterializedValue(_ => NotUsed)
   }
 
+  def persistenceIds(entityType: String, afterId: Option[String], limit: Long): Source[String, NotUsed] = {
+    val result = r2dbcExecutor.select(s"select persistenceIds by entity type")(
+      connection =>
+        afterId match {
+          case Some(after) =>
+            connection
+              .createStatement(persistenceIdsForEntityTypeAfterSql)
+              .bind(0, entityType + "%")
```

Comment:
Should it include the `|` separator, just in case one entity type is a prefix for another?

```scala
entityType + "|%"
```

Reply:
I didn't add "|" because of this construction method `def of(entityTypeHint: String, entityId: String, separator: String): PersistenceId =` I could add an additional parameter to `def persistenceIds(` to provide the separator. @patriknw   `only supports the default separator` how do we block using different separators in r2dbc?

Reply:
As said, r2dbc only supports the default separator. Supporting custom separator would be a different separate issue (big task with little value). Created issue https://github.com/akka/akka-persistence-r2dbc/issues/347

I don't think we have anything checks that the `persistenceId: String` must contain the default separator, so it will probably work to write the events, but some queries that rely on entityType might not work. I think that is fine for this new query as well since that is already the case in many other places.

Reply:
Hmm, so potentially a user could pick another separator when defining their `PersistenceId` and get it all broken? Do we have this documented?

Reply:
Ok, so it's by the convention. I could write some java/scala doc comments that the `currentPersistenceIds` method will work only with a default separator.

Reply:
Added.